acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -372,13 +372,15 @@ def connect_pcp_client(broker)
     })
     connected = client.connect(5)
     retries += 1
+
+    # If the connection was not established, close it. Otherwise we can end up with 2 successful connections
+    # and get an earlier attempt superseding later connections.
+    client.close if (!connected || !client.associated?)
```

Comment:
Actually, it looks like `connect()` will return `nil` if association times out. So closing the connection on `!connected` alone seems sufficient.